pilot/pkg/credentials/kube/multicluster.go

```diff
@@ -146,3 +146,19 @@ func (a *AggregateController) Authorize(serviceAccount, namespace string) error
 func (a *AggregateController) AddEventHandler(f func(name string, namespace string)) {
 	// no ops
 }
+
+func (a *AggregateController) GetDockerCredential(name, namespace string) ([]byte, error) {
+	// Search through all clusters, find first non-empty result
```

Comment:
I wonder if we should make this a bit stricter and make it only the same cluster as the requesting proxy?

A downside is it is possibly more work to have someone replicate this. The positive is its less permissive, matches K8s image pulling. Also the downside, IMO, is not that bad since they already need to replicate it to each NS

Reply:
Actually nvm, noticed we actually allow pulling root credentials directly

Reply:
And looks like we do call `secretController.ForCluster(proxy)` already